src/Components/FitnessPage.js

```diff
@@ -0,0 +1,280 @@
+    const isCurrent = index == activeIndex;
+
+    const doneStyle = index < activeIndex ? styles.stepTitleDone : {};
+    const currentStyle = isCurrent ? styles.stepTitleCurrent : {};
+    const currentStep = isCurrent ? styles.stepCurrent : {};
+
+    return (
+      <View style={styles.page}>
+        <TouchableOpacity style={[styles.step, currentStep]} onPress={() => setIndex(index)}>
+          <View style={styles.subWrapper}>
+            <Text style={[styles.stepTitle, doneStyle, currentStyle]}>{item.repetition} </Text>
+            <Text style={[styles.stepTitle, doneStyle, currentStyle]}>{item.title}</Text>
+          </View>
+        </TouchableOpacity>
+      </View>
+    );
+  };
+
+  const renderVimeo = ({ vimeoId, sounds }) => {
+    const isUniqueStep = stepsCount === 1;
+    return (
+      <>
+        <View style={styles.allPage}>
+          <View style={[styles.videoMultiple, isUniqueStep && styles.videoAlone]}>
+            <VimeoVideo vimeoId={vimeoId} sounds={sounds} shouldPlay />
+          </View>
+          <ScrollView style={styles.scrollList}>
+            {!isUniqueStep && (
+              <FlatList
+                nestedScrollEnabled
+                ref={flatListRef}
+                data={drill.steps}
+                keyExtractor={(item) => item.id.toString()}
+                renderItem={renderStep}
+              />
+            )}
+          </ScrollView>
+        </View>
+      </>
+    );
+  };
+
+  const renderAnimation = ({ animation, title, instruction }, index) => {
+    const isFirstStep = index === 0;
+    const isLastStep = index === stepsCount - 1;
+    return (
+      <>
+        <View style={styles.line}>
+          <View>
+            {!isFirstStep && (
+              <TouchableOpacity
+                onPress={() => {
+                  carouselRef.current.snapToPrev();
+                }}
+              >
+                <MaterialCommunityIcons name="chevron-double-left" color={theme.COLOR_PRIMARY} size={26} />
+              </TouchableOpacity>
+            )}
+          </View>
+          <Text style={styles.title}>{title}</Text>
+          <View>
+            {!isLastStep && (
+              <TouchableOpacity
+                onPress={() => {
+                  carouselRef.current.snapToNext();
+                }}
+              >
+                <MaterialCommunityIcons name="chevron-double-right" color={theme.COLOR_PRIMARY} size={26} />
+              </TouchableOpacity>
+            )}
+          </View>
+        </View>
+        <View style={styles.animation}>
+          <Animation widthRatio={1} heightRatio={1 / 2} animation={new Drill(animation)} />
+        </View>
+        <Text style={styles.instruction}>{instruction}</Text>
+      </>
+    );
+  };
+
+  return (
+    <View style={styles.container}>
+      {activeIndex === drill.steps.length && renderFinish()}
+      {currentStep?.animation && !currentStep?.vimeoId && renderAnimation(currentStep)}
```

Comment:
If this is a complete copy-paste from FitnessDrillIllustration, why don't use it instead of copy-pasting ?

Reply:
Le style n'est plus exactement le même. Couleur et rond pour checker